File: PregnancyAppBackend/Dtos/Web/DailyHealthPrediction/DailyHealthPredictionResultDto.cs
namespace PregnancyAppBackend.Dtos.Web.DailyHealthPrediction;

public class DailyHealthPredictionResultDto
{
    public Guid DailySurveyId { get; set; }
    public DateTime CreationDateUtc { get; set; }

    public int Trimester { get; set; }
    public int Prediction { get; set; }
    public string PredictionText { get; set; } = string.Empty;

    public PredictionProbabilitiesDto Probabilities { get; set; } = new();
    public List<DailyHealthPredictionDeviationDto> Deviations { get; set; } = new();
}

public class PredictionProbabilitiesDto
{
    public double Normal { get; set; }
    public double Alert { get; set; }
    public double Pathology { get; set; }
}

public class DailyHealthPredictionDeviationDto
{
    public string Feature { get; set; } = string.Empty;
    public decimal Value { get; set; }

    public string NormalRange { get; set; } = string.Empty;
    public string Severity { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
}

File: PregnancyAppFrontend/src/app/dtos/daily-health-prediction/daily-health-prediction-result-dto.ts
export interface PredictionProbabilitiesDto {
  normal: number;
  alert: number;
  pathology: number;
}

export interface DailyHealthPredictionDeviationDto {
  feature: string;
  value: number;
  normalRange: string;
  severity: string;
  message: string;
}

export interface DailyHealthPredictionResultDto {
  dailySurveyId: string;
  creationDateUtc: string; // ISO
  trimester: number;
  prediction: number; // 0=Норма,1=Предупреждение,2=Патология
  predictionText: string;
  probabilities: PredictionProbabilitiesDto;
  deviations: DailyHealthPredictionDeviationDto[];
}
